src/URDFConverter/UrdfUnity/UrdfUnity/FileManager/FileManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.IO;
+using UrdfUnity.Urdf.Models;
+using UrdfUnity.Util;
+using UrdfUnity.Parse;
+using NLog;
+
+
+namespace UrdfUnity.FileManager
+{
+    /// <summary>
+    /// Filetypes we care about in the file manager.
+    /// </summary>
+    public enum FileType
+    {
+        UNKNOWN,
+        URDF,
+        XACRO
+    }
+
+
+    /// <summary>
+    /// This class will be handling the FileIO 
+    /// </summary>
+    public class FileManager
+    {
+        Logger LOGGER;
+        
+
+        /// <summary>
+        /// Base constructor, for initializing components
+        /// </summary>
+        public FileManager()
+        {
+            // initializing the logger for this class.
+            LOGGER = LogManager.GetCurrentClassLogger();
+        }
+
+
+        /// <summary>
+        /// We have a URDF/XACRO filepath that needs to be extracted to be
+        /// a Robot model. 
+        /// </summary>
+        /// <param name="filePath">Path of file containing urdf/xacro for the robot</param>
+        /// <returns>
+        ///     Robot|null - A Robot model if a valid file path and content for a robot. 
+        ///     A null returned when invalid filepath or content is supplied.
+        /// </returns>
+        public Robot GetRobotFromFile(string filePath)
+        {
+            Robot robo = null;
+            string fileString = "";
+            try
+            {
+                using (StreamReader fp = new StreamReader(filePath))
+                {
+                    fileString = fp.ReadToEnd();
+                }
+            }
+            catch (FileNotFoundException fnf_ex)
+            {
+                LOGGER.Error("File not found \"{0}\".");
+            }
+
+
+            if (!fileString.Equals(string.Empty))
+            {
+                FileType type = GetFileType(filePath);
+                switch(type)
+                {
+                    case FileType.URDF:
+                        UrdfParser urdf = new UrdfParser();
```

Comment:
Let's make this a class member called urdfParser, since it can be re-used (and use this. to access)